static/src/javascripts/projects/common/modules/ui/subscription-banner.js

```diff
@@ -0,0 +1,163 @@
+};
+
+const subcriptionBannerCloseActions = () => {
+    closedAt(subsciptionBannerClosedKey);
+    bannerHasBeenAcknowledged();
+};
+
+const onAgree = (): void => {
+    allAdConsents.forEach(_ => {
+        setAdConsentState(_, true);
+    });
+};
+
+const bindCloseHandler = (button, banner, callback) => {
+    if (button) {
+        button.addEventListener('click', () => {
+            callback();
+            banner.remove();
+        });
+    }
+};
+
+const bindClickHandler = (button, callback) => {
+    if (button) {
+        button.addEventListener('click', () => {
+            callback();
+        });
+    }
+};
+
+const trackSubscriptionBannerView = () => {
+    submitViewEvent({
+        component: {
+            componentType: 'ACQUISITIONS_OTHER',
+            id: 'acquisitions-subscription-banner',
+        },
+    });
+};
+
+const trackSubscriptionBannerCtaClick = () => {
+    submitClickEvent({
+        component: {
+            componentType: 'ACQUISITIONS_OTHER',
+            id: 'acquisitions-subscription-banner',
+        },
+    });
+};
+
+const bindSubscriptionClickHandlers = () => {
+    const subscriptionBannercloseButton = document.querySelector(
+        '#js-site-message--subscription-banner__cta-dismiss'
+    );
+    const subscriptionBannerHtml = document.querySelector(
+        '#js-subscription-banner-site-message'
+    );
+
+    const subscriptionBannerCta = document.querySelector(
+        '#js-site-message--subscription-banner__cta'
+    );
+
+    if (subscriptionBannerHtml) {
+        bindCloseHandler(
+            subscriptionBannercloseButton,
+            subscriptionBannerHtml,
+            subcriptionBannerCloseActions
+        );
+        bindClickHandler(
+            subscriptionBannerCta,
+            trackSubscriptionBannerCtaClick
+        );
+    }
+};
+
+const bindConsentClickHandlers = () => {
+    const consentBannerCloseButton = document.querySelector(
+        '.site-message--first-pv-consent__button'
+    );
+    const consentBannerHtml = document.querySelector(
+        '#js-first-pv-consent-site-message'
+    );
+
+    if (consentBannerHtml) {
+        bindCloseHandler(consentBannerCloseButton, consentBannerHtml, onAgree);
+    }
+};
+
+const show: () => Promise<boolean> = () => {
+    trackFirstPvConsent();
+    trackSubscriptionBannerView();
+
+    if (document.body) {
+        document.body.insertAdjacentHTML(
+            'beforeend',
+            bannerTemplate(subscriptionUrl, signInUrl)
+        );
+    }
+
+    bindConsentClickHandlers();
+    bindSubscriptionClickHandlers();
+
+    return Promise.resolve(true);
+};
+
+const canShow: () => Promise<boolean> = () => {
+    const can = Promise.resolve(
+        fiveOrMorePageViews(pageviews) &&
+            !hasUserAcknowledgedBanner(messageCode) &&
+            !isAustralianEdition(edition) &&
+            !shouldHideSupportMessaging() &&
+            !pageShouldHideReaderRevenue() &&
+            subscriptionBannerSwitch
```

Comment:
Nitpick: `subscriptionBannerSwitchIsOn`?

Reply:
changed to subscriptionBannerSwitchIsOn